serving/using-cert-manager-on-gcp.md

```diff
@@ -0,0 +1,228 @@
+
+```shell
+kubectl get clusterissuer -n cert-manager letsencrypt-issuer -o yaml
+```
+
+and confirm that its conditions have `Ready=True`.  For an example:
+
+```yaml
+status:
+  acme:
+    uri: https://acme-v02.api.letsencrypt.org/acme/acct/40759665
+  conditions:
+  - lastTransitionTime: 2018-08-23T01:44:54Z
+    message: The ACME account was registered with the ACME server
+    reason: ACMEAccountRegistered
+    status: "True"
+    type: Ready
+```
+
+### Specifying our certificate
+
+Next we need to configure which certificate issuer to use
+and which secret we'll publish the certificate into.
+
+```shell
+# Change this value to the domain you want to use.
+export DOMAIN=your-domain.com
+
+kubectl apply -f - <<EOF
+apiVersion: certmanager.k8s.io/v1alpha1
+kind: Certificate
+metadata:
+  name: my-certificate
+  # Istio certs secret lives in the istio-system namespace, and
+  # a cert-manager Certificate is namespace-scoped.
+  namespace: istio-system
+spec:
+  # Reference to the Istio default cert secret.
+  secretName: istio-ingressgateway-certs
+  acme:
+    config:
+    # Each certificate could comprise of different ACME challenge
+    # solver.  In this example we are using one provider for all
+    # the domains.
+    - dns01:
+        provider: cloud-dns-provider
+      domains:
+      # Since certificate wildcards only allow one level, we will
+      # need to one for every namespace.  We don't need to use
+      # wildcard here, fully-qualified domains will work fine too.
+      - "*.default.$DOMAIN"
+      - "*.other-namespace.$DOMAIN"
+  # The certificate common name, use one from your domains.
+  commonName: "*.default.$DOMAIN"
+  dnsNames:
+  - "*.default.$DOMAIN"
+  - "*.other-namespace.$DOMAIN"
+  # Reference to the ClusterIssuer we created in the previous step.
+  issuerRef:
+    kind: ClusterIssuer
+    name: letsencrypt-issuer
+EOF
+```
+
+To check that your certificate setting is valid, run
+
+```shell
+kubectl get certificate -n istio-system my-certificate -o yaml
+```
+
+and verify that its `Status.Conditions` have `Ready=True`.  For an example
+
+```yaml
+status:
+  acme:
+    order:
+      url: https://acme-v02.api.letsencrypt.org/acme/order/40759665/45358362
+  conditions:
+  - lastTransitionTime: 2018-08-23T02:28:44Z
+    message: Certificate issued successfully
+    reason: CertIssued
+    status: "True"
+    type: Ready
+```
+
+### Configure our Gateway 
+
+In the last step, we configure the knative-shared-gateway to use the certificate
+that is generated and stored automatically by cert-manager.
+
+The key edit here is adding the `tls:` section to the end of the HTTPS port
+configuration.
+
+```shell
+kubectl apply -f - <<EOF
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: knative-shared-gateway
+  namespace: knative-serving
+spec:
+  selector:
+    knative: ingressgateway
+  servers:
+  - port:
+      number: 80
+      name: http
+      protocol: HTTP
+    hosts:
+    - "*"
```

Comment:
Should this be set to redirect users to https? (In a different PR)

Reply:
Done